eng/steps/build-test.yaml

```diff
@@ -94,11 +95,10 @@ steps:
       testResultsFiles: ${{ parameters.GoTestPath }}/**/report.xml
       failTaskOnFailedTests: true
 
-  - ${{if ne(variables['Build.Reason'], 'PullRequest')}}:
+  - ${{if and(ne(variables['Build.Reason'], 'PullRequest'), eq(parameters.PublishAutorestCoverage, true))}}:
       - pwsh: |
           $currentVersion = node -p -e "require('./src/package.json').version";
           $currentVersion="$currentVersion-$(Build.BuildNumber)";
           cd src/node_modules/@microsoft.azure/autorest.testserver;
-          npm run coverage -- publish --repo=$(Build.Repository.Name) --ref=$(Build.SourceBranch) --githubToken=$(azuresdk-github-pat) --azStorageAccount=$(storage-coverage-user) --azStorageAccessKey=$(storage-coverage-pass) --version=$currentVersion;
+          npm run coverage -- publish --repo=$(Build.Repository.Name) --ref=$(Build.SourceBranch) --githubToken=$(github-token) --azStorageAccount=$(storage-coverage-user) --azStorageAccessKey=$(storage-coverage-pass) --version=$currentVersion;
```

Comment:
Ben fixed this yesterday, did you mean to revert it?

Reply:
Yes. `github-token`, `storage-coverage-user` and `storage-coverage-pass`is configured in the public ci variable. This step should not be executed in private ci (publish npm package).

Reply:
Shouldn't we skip publish altogether on the public pipeline? We don't want any credentials exposed for that since any pull request can access it automatically.

Reply:
> Shouldn't we skip publish altogether on the public pipeline? We don't want any credentials exposed for that since any pull request can access it automatically.

The value set to the variable `github-token` in public repo is `skip`. This pipeline was set up long time ago and I just did some refactor. It seems the common pipeline for all languages' codegen.